receiver/datadogreceiver/metrics_translator.go

```diff
@@ -13,12 +15,104 @@ import (
 
 type MetricsTranslator struct {
 	sync.RWMutex
-	buildInfo component.BuildInfo
-	lastTs    map[identity.Stream]pcommon.Timestamp
+	buildInfo  component.BuildInfo
+	lastTs     map[identity.Stream]pcommon.Timestamp
+	stringPool *StringPool
 }
 
 func newMetricsTranslator() *MetricsTranslator {
 	return &MetricsTranslator{
-		lastTs: make(map[identity.Stream]pcommon.Timestamp),
+		lastTs:     make(map[identity.Stream]pcommon.Timestamp),
+		stringPool: newStringPool(),
 	}
 }
+
+func (mt *MetricsTranslator) streamHasTimestamp(stream identity.Stream) (pcommon.Timestamp, bool) {
+	mt.RLock()
+	defer mt.RUnlock()
+	ts, ok := mt.lastTs[stream]
+	return ts, ok
+}
+
+func (mt *MetricsTranslator) updateLastTsForStream(stream identity.Stream, ts pcommon.Timestamp) {
+	mt.Lock()
+	defer mt.Unlock()
+	mt.lastTs[stream] = ts
+}
+
+const (
+	TypeGauge string = "gauge"
+	TypeRate  string = "rate"
+	TypeCount string = "count"
+)
+
+type SeriesList struct {
+	Series []datadogV1.Series `json:"series"`
+}
+
+func translateMetricsV1(series SeriesList, mt *MetricsTranslator) pmetric.Metrics {
+	bt := newBatcher()
+	bt.Metrics = pmetric.NewMetrics()
+
+	for _, serie := range series.Series {
+		var dps pmetric.NumberDataPointSlice
+
+		dimensions := parseSeriesProperties(serie.Metric, serie.GetType(), serie.GetTags(), serie.GetHost(), mt.buildInfo.Version, mt.stringPool)
+		metric, metricID := bt.Lookup(dimensions)
+
+		switch serie.GetType() {
+		case TypeCount:
+			metric.Sum().SetAggregationTemporality(pmetric.AggregationTemporalityDelta)
+			metric.Sum().SetIsMonotonic(false) // See https://docs.datadoghq.com/metrics/types/?tab=count#definition
+			dps = metric.Sum().DataPoints()
+		case TypeGauge:
+			dps = metric.Gauge().DataPoints()
+		case TypeRate:
+			metric.Sum().SetAggregationTemporality(pmetric.AggregationTemporalityDelta)
+			dps = metric.Sum().DataPoints()
+		default:
+			// Type is unset/unspecified
```

Comment:
lets record the dp type as a counter (count/gauge/rate/unknown). can be done in a later pr

Reply:
Are you talking about the entire switch case? I believe its better if metrics are properly identified from the start.

Reply:
oh I meant it would be nice to have this receiver itself export a metric like `otelcol_datadog_series_total{type="count|gauge|rate|undefined"}`